impl/maven-di/src/main/java/org/apache/maven/di/Injector.java

```diff
@@ -161,4 +161,16 @@ static Injector create() {
      */
     @Nonnull
     <T> T getInstance(@Nonnull Key<T> key);
+
+    /**
+     * Disposes this Injector, clearing all internal state (bindings, caches, scopes, etc.).
+     * After calling this, the Injector should not be used again.
+     * @since 4.1
+     */
+    default void dispose() {
+        // delegate to the implementation
+        if (this instanceof InjectorImpl) {
```

Comment:
I think we can use `if (this instanceof InjectorImpl impl)` and avoid the cast below.